docs/arguments.md

```diff
@@ -1,5 +1,57 @@
 # Cortex Arguments Explained
 
+## Querier
+
+- `-querier.max-concurrent`
+
+   The maximum number of top-level PromQL queries that will execute at the same time, per querier process.
+
+- `-querier.query-parallelism`
+
+   This refers to database queries against the store (e.g. Bigtable or DynamoDB).  This is the max subqueries run in parallel per higher-level query.
```

Comment:
Ah, the flag description is misleading.  "Sub-query" means promql query, but its subquery in that a single promql query is sharded into multiple single day queries.

This should be set to `max-concurrent / # query-frontend replicas`, as this is the number of goroutines which connect to each frontend in parallel.  If you have `query-parallelism > (max-concurrent / # query-frontend replicas)`, then you'll have queries queueing in the queriers and not the frontend, which will affect QoS.

Has no effect if you're not running the query-frontend.

Reply:
I'm wrong here...